terraform/eks/otlp/push_mode_samples_inject_auto_instrumentation.tf

```diff
@@ -0,0 +1,96 @@
+# ------------------------------------------------------------------------
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+##########################################
+# Push mode deployment to inject with auto_instrumentation
+##########################################
+
+# deploy sample app
+resource "kubernetes_deployment" "push_mode_sample_app_deployment_inject_auto_instrumentation" {
```

Comment:
@bryan-aguilar I tried that initially but the environment variables and annotations required in these two cases are different. 
For auto-instrumentation through Operator, I preferred to inject OTLP specific env variables through the Operator Instrumentation template file. And the test-case fails if we use the `push_mode_samples.tf` instead of using `push_mode_samples_inject_auto_instrumentation`, b/c of the difference in env variables.

Reply:
@bryan-aguilar I can consolidate the two sample app deployment into one. But then it's not possible to pass the exporter endpoints through the `java_auto_instrumentation.tpl`, because the `OTEL_EXPORTER_OTLP_ENDPOINT` env variable gets set in `push_mode_samples.tf` and I couldn't get the Operator to override the env values provided with sample_app deployment.  
I thought the right way for this test, is to provide all the OTLP related env variables through the Instrumentation template and not through sample app deployment. And for this reason created a new deployment for sample app without env variables. 
Do you want me to consolidate the apps deployment and keep the env variables set through the app deployment?

Reply:
@bryan-aguilar consolidated `push_mode_samples.tf`.

Reply:
What was your strategy for resolving your previous concerns? I don't see any changes related to the environment variables provided to the deployment.

Reply:
@bryan-aguilar : I am not providing the OTLP related env variables through the `kind:Instrumentation` template now. By not doing that, I was able to consolidate the workload deployment templates.

Reply:
@bryan-aguilar yes its a trade off ( I couldn't find a way to override env variables already provided at workload deployment stage). but also discussed this with @rapphil, he also mentioned that since priority of this test is not to test whether the env variables to the workload can be set through the instrumentation crd template - so we should consolidate the workload deployment template to increase readability.